### PR TITLE
Add plugin docs to website

### PR DESCRIPTION
**Pull request by samwestmoreland**

### CODE REVIEW

File: docs/BUILD [hunk 1]
```diff
@@ -52,6 +53,48 @@ genrule(
     },
 )
 
+plugins = {
+    "python": "v1.2.0",
+    "java": "v0.2.5",
+    "go": "v1.1.4",
+    "cc": "v0.3.2",
+    "shell": "v0.1.1",
+}
```
[the remark this answers is not in the document]
Comment (samwestmoreland) [reply docs/BUILD:64]: Yeah, I wasn't sure what to do about that. Linting's a good idea! I'll see if I can work it out.

File: docs/build_defs/docs.build_defs [hunk 2]
```diff
@@ -23,4 +23,33 @@ def html_template(name : str, src : str, output : str = ''):
...
+    )
+
+    json = build_rule(
+        name = name,
+        srcs = [extracted],
+        outs = [f"{name}_rules.json"],
+        tag = 'json',
+        cmd = "$TOOL init && $TOOL query rules $SRCS/* > $OUT",
+        tools = ["//src:please"],
+    )
```
[the remark this answers is not in the document]
Comment (samwestmoreland) [reply docs/build_defs/docs.build_defs:50]: Without `plz init` `plz query rules` doesn't work. Just produces an empty json... Not exactly sure why

File: docs/tools/lexicon_templater/template_lexicon.go [hunk 3]
```diff
@@ -59,18 +61,28 @@ func must(err error) {
...
+	Input []string `short:"i" long:"input" description:"Input file(s)"`
+	Rules []string `short:"r" long:"rules" description:"Rules file(s)"`
+}
+
 func main() {
+	flags.ParseFlagsOrDie("Docs template", &opts, nil)
 	r := &rules{}
-	tmpl, err := template.New("lexicon.html").Funcs(template.FuncMap{
+	split := strings.Split(opts.Input[0], "/")
+	basename := split[len(split)-1]
```
[the remark this answers is not in the document]
Comment (samwestmoreland) [reply docs/tools/lexicon_templater/template_lexicon.go]: Ah good point!

**Review by Tatskaari — APPROVED** [review]

**Review by Tatskaari — APPROVED** [review]
Nice! Looks good!

**Pull request merged**